src/app/views/impac/impac.controller.coffee

```diff
@@ -5,11 +5,17 @@ angular.module 'mnoEnterpriseAngular'
     #====================================
     # Post-Initialization
     #====================================
-    $scope.$watch MnoeOrganizations.getSelectedId, (newValue, oldValue) ->
-      if newValue? && oldValue?
-        console.log "Reload with id", newValue
 
-        # Reload the dashboard
-        ImpacDashboardsSvc.load(true)
+    # Triggers a dashboard reload with a loader spinner.
+    reloadImpacDashboard = ->
+      ImpacDashboardsSvc.triggerDhbLoader(true)
+      ImpacDashboardsSvc.load(true).finally(->
+        ImpacDashboardsSvc.triggerDhbLoader(false)
+      )
+
+    # Watches for organization selector drop-down change
+    $scope.$watch(MnoeOrganizations.getSelectedId, (newValue, oldValue) ->
```

Comment:
that's correct, but I think it will still be called every time the `selectedId` changes (eg: when you switch items in the left-side menu), so I'm not sure your problem is solved?.. :/